### PR TITLE
Fixed an export problem when moving tensors to CPU during `torch.export.save`

### LINKED ISSUE #113999: [export] VGG model export.load() failure
### 🐛 Describe the bug

With recent torch nightly, torch.export.load() on torchvision vgg 16 fails with the following error

```
torch._export.verifier.SpecViolationError: State dict entry for parameter features.0.weight is not an instance of torch.nn.Parameter.
```

To reproduce 
```
import torch
import torchvision.models as models 

model = models.vgg16(weights=None).eval().cuda()
random_inputs = [torch.rand([32, 3, 32, 32]).to("cuda")]
dim_x = torch.export.Dim("dim_x", min=1, max=32)
exp_program = torch.export.export(model, tuple(random_inputs), dynamic_shapes={"x" : {0 : dim_x}})
torch.export.save(exp_program, "./vgg_test.ep")
des_ep = torch.export.load("vgg_test.ep")
print(des_ep.module())
```

### Versions

Versions of relevant libraries:
[pip3] mypy==1.5.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.25.2
[pip3] onnx==1.14.1
[pip3] pytorch-quantization==2.1.2
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] pytorch-triton==2.1.0+6e4932cda8
[pip3] torch==2.2.0.dev20231116+cu121
[pip3] torch-tensorrt==2.2.0.dev0+7ad927248
[pip3] torchvision==0.17.0.dev20231116+cu121
[pip3] triton==2.1.0

cc @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### PR DESCRIPTION
**Pull request by tringwald**
For whatever reason calling`.cpu()` on a `nn.Parameter` wrapping a CUDA tensor will return a plain (non-parameter) tensor. This PR fixes the symptom in the linked issue, but not the underlying issue.

Fixes #113999.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### CODE REVIEW

**Review by zhxchen17 — APPROVED** [review]

**Review by zhxchen17 — CHANGES_REQUESTED** [review]
some question regarding the added test.

File: test/dynamo/test_export.py [hunk 1]
```diff
@@ -32,6 +33,15 @@
     ShapeEnv,
 )
 from torch.testing._internal import common_utils
+from torch.testing._internal.common_cuda import TEST_CUDA
+
+try:
+    import torchvision.models
```
Comment (zhxchen17) [inline test/dynamo/test_export.py:39]: @tringwald sorry I was aware of this when reviewed it for the first time. Do you mind creating a repro without depending on torchvision? I think it's better to have tests always enabled so we can make sure something is regressed on all diffs.

Comment (tringwald) [reply test/dynamo/test_export.py:39]: Sure, I'll try to find a repro without using torchvision. The CUDA part seems necessary though, I wasn't able to reproduce it on CPU.

**Review by zhxchen17 — APPROVED** [review]
@tringwald thanks!

**Pull request closed without merging**